packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -179,17 +186,43 @@ class CupertinoSwitch extends StatefulWidget {
   /// See also:
   ///
   ///  * [trackColor], the color to use for the track when the switch is off.
+  @Deprecated('Please use activeTrackColor')
   final Color? activeColor;
 
+  final Color? _activeTrackColor;
+  /// The color to use for the track when the switch is on.
+  ///
+  /// If null and [applyTheme] is false, defaults to [CupertinoColors.systemGreen]
+  /// in accordance to native iOS behavior. Otherwise, defaults to
+  /// [CupertinoThemeData.primaryColor].
+  ///
+  /// See also:
+  ///
+  ///  * [_activeTrackColor], the color to use for the track when the switch is off.
+  Color? get activeTrackColor => _activeTrackColor;
```

Comment:
Piggybacking off of @Piinks's review and comments, it would be better if this was a normal property and not a getter method. Is it a getter now so that code outside the widget is able to get the correct color in a backwards compatible way? We should have `activeColor` do that, combined with the deprecation message it will direct them to the right property while giving them the right value. In the future, there's no need for `activeTrackColor` to be a getter.